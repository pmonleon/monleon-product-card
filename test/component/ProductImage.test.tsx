import React from 'react';
import render  from 'react-test-renderer'
// import * as ReactDOM from 'react-dom';
import { ProductCard, ProductImage } from '../../src/components';
import { products } from './data/products';

const mockProducts = products

describe('ProducImage', () => { 

    test('Debe mostrar el componente Title con los dtaos de las props', () => { 
        const wrapper1 = render.create(
            <ProductImage
                img='./no-image.jpg'
                className='custom-class'
            />
        )
        
       // console.log(wrapper.toJSON())
        expect(wrapper1.toJSON()).toMatchSnapshot()
        
        
    })

    test('Debe mostrar el componente Title con los datos del context', () => { 
        const wrapper = render.create(
            <ProductCard product={mockProducts[1]}>
                {
                    (  ) => (
                        <ProductImage/>
                        )
                    }
            </ProductCard>
        )
        
        expect(wrapper.toJSON()).toMatchSnapshot()
     })
})
