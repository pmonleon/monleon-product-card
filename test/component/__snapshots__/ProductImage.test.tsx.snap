// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProducImage Debe mostrar el componente Title con los datos del context 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProducImage Debe mostrar el componente Title con los dtaos de las props 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="./no-image.jpg"
/>
`;
